libarchive/archive_read_support_format_cpio.c

```diff
@@ -1030,6 +1032,12 @@ atol8(const char *p, unsigned char_cnt)
 
 static int64_t
 atol16(const char *p, unsigned char_cnt)
+{
```

Comment:
`atol16` was already parsing an unsigned value and then casting to signed.  So we can easily split it into `atol16u` that parses an unsigned value, and `atol16` that casts to signed.